src/modules/site-v2/base/views/tools/phenotype_comparison.py

```diff
@@ -0,0 +1,138 @@
+import bleach
+from caendr.services.logger import logger
+from flask import Response, Blueprint, render_template, request, url_for, jsonify, redirect, flash, abort
+
+from base.forms              import PhenotypeComparisonForm
+from base.utils.auth         import jwt_required, get_current_user, user_is_admin
+from base.utils.tools        import lookup_report, try_submit
+
+from caendr.models.datastore import Species, PhenotypeReport
+from caendr.models.error     import ReportLookupError, EmptyReportDataError, EmptyReportResultsError
+from caendr.models.status    import JobStatus
+
+
+
+phenotype_comparison_bp = Blueprint(
+  'phenotype_comparison', __name__, template_folder='templates'
+)
+
+
+
+@phenotype_comparison_bp.route('/')
+def phenotype_comparison():
+
+  # Construct variables and render template
+  return render_template('tools/phenotype/submit.html', **{
+
+    # Page info
+    "title": "Phenotype Comparison",
+    "form":  PhenotypeComparisonForm(request.form),
+    "tool_alt_parent_breadcrumb": {
+      "title": "Tools",
+      "url":   url_for('tools.tools')
+    },
+
+    # List of Species class fields to expose to the template
+    "species_list": Species.all(),
+    'species_fields': [
+      'name', 'short_name', 'project_num', 'wb_ver', 'release_latest',
+    ],
+
+    # String replacement tokens
+    # Maps token to the field in Species object it should be replaced with
+    'tokens': {
+      'WB':      'wb_ver',
+      'RELEASE': 'release_latest',
+      'PRJ':     'project_num',
+      'GENOME':  'fasta_genome',
+    },
+
+    # Misc
+    "fluid_container": True,
+  })
+
+
+
+@phenotype_comparison_bp.route('/submit', methods=["POST"])
+@jwt_required()
+def submit():
+  form = PhenotypeComparisonForm(request.form)
+  user = get_current_user()
+
+  # Validate form fields
+  # Checks that species is in species list & label is not empty
+  if not form.validate_on_submit():
+    msg = "Invalid submission"
```

Comment:
if we can be more descriptive to the end-user on specifics on why the submission failed, we should consider it.

Reply:
For our records -- this method of receiving / validating the submission has been replaced with a JSON solution.  Will still be worth testing invalid submission values.

See [current solution](https://github.com/AndersenLab/CAENDR/blob/191763d0cb7a9284d074735a545c3c02dd732c59/src/modules/site-v2/base/views/tools/phenotype_database.py#L121) (as of this comment).